types/mapcmd/map.go

```diff
@@ -27,10 +28,36 @@ import (
 
 func New(config *hazelcast.Config) *cobra.Command {
 	var cmd = &cobra.Command{
-		Use:   "map {get | put} --name mapname --key keyname [--value-type type | --value-file file | --value value]",
-		Short: "Map operations",
+		Use:     "map {get | put} --name mapname --key keyname [--value-type type | --value-file file | --value value]",
+		Short:   "Map operations",
+		Example: fmt.Sprintf("%s\n%s", MapPutExample, MapGetExample),
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// All the following lines are to set map name if it is set by "use" command.
+			// If the map name is given explicitly, do not set the one given with "use" command.
+			// Missing flag errors are not handled here.
+			// They are expected to be handled by the actual command.
+			persister := internal.PersisterFromContext(cmd.Context())
+			val, isSet := persister.Get("map")
+			if !isSet {
+				return nil
+			}
+			nameFlag := cmd.Flag("name")
+			if nameFlag == nil {
+				// flag is absent
+				return nil
+			}
+			if nameFlag.Changed {
+				// flag value is set explicitly
+				return nil
+			}
+			if err := cmd.Flags().Set("name", val); err != nil {
+				cmd.PrintErrln("cannot set persistent err", err)
```

Comment:
How can this error message get triggered?

Reply:
If no such flag ("name" in our case) is defined or it is deprecated. There may be couple more reasons.

Reply:
But in the code above we check whether the `name` flag is defined?
```
nameFlag := cmd.Flag("name")
if nameFlag == nil {
    // flag is absent
    return nil
}
```
Is the user supposed to receive "cannot set persistent err ...". I don't know what it means.

Reply:
Does the message for the loggable error below get printed too?

Reply:
What is your proposal? Should we ignore the error? I think it is good to have it printed, it lets  us catch it before we can release it. Wdyt?

Reply:
Ah I see, redundant log is removed on https://github.com/hazelcast/hazelcast-commandline-client/pull/61/commits/c3073e7853d182ba40c0a909739f23428c169e74